Venmo/src/test/java/com/braintreepayments/api/VenmoClientUnitTest.java

```diff
@@ -93,15 +98,96 @@ public void showVenmoInGooglePlayStore_sendsAnalyticsEvent() {
     }
 
     @Test
-    public void tokenizeVenmoAccount_launchesVenmoWithCorrectVenmoExtras() throws JSONException, InvalidArgumentException {
+    public void tokenizeVenmoAccount_createsPaymentContextViaGraphQL() throws JSONException {
         BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
                 .configuration(venmoEnabledConfiguration)
                 .sessionId("session-id")
                 .integration("custom")
                 .authorization(Authorization.fromString(base64Encode(Fixtures.CLIENT_TOKEN)))
                 .build();
 
-        VenmoRequest request = new VenmoRequest();
+        when(deviceInspector.isVenmoAppSwitchAvailable(activity)).thenReturn(true);
+
+        VenmoRequest request = new VenmoRequest(VenmoPaymentMethodUsage.SINGLE_USE);
+        request.setProfileId("sample-venmo-merchant");
+        request.setShouldVault(false);
+        request.setPaymentMethodUsage(VenmoPaymentMethodUsage.SINGLE_USE);
+
+        VenmoClient sut = new VenmoClient(braintreeClient, tokenizationClient, sharedPrefsWriter, deviceInspector);
+        sut.tokenizeVenmoAccount(activity, request, venmoTokenizeAccountCallback);
+
+        ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+        verify(braintreeClient).sendGraphQLPOST(captor.capture(), any(HttpResponseCallback.class));
+
+        String graphQLBody = captor.getValue();
+        JSONObject graphQLJSON = new JSONObject(graphQLBody);
+
+        String expectedQuery = "mutation CreateVenmoPaymentContext($input: CreateVenmoPaymentContextInput!) { createVenmoPaymentContext(input: $input) { venmoPaymentContext { id } } }";
+        assertEquals(expectedQuery, graphQLJSON.getString("query"));
+
+        JSONObject variables = graphQLJSON.getJSONObject("variables");
+        JSONObject input = variables.getJSONObject("input");
+        assertEquals("SINGLE_USE", input.getString("paymentMethodUsage"));
+        assertEquals("sample-venmo-merchant", input.getString("merchantProfileId"));
+        assertEquals("MOBILE_APP", input.getString("customerClient"));
+        assertEquals("CONTINUE", input.getString("intent"));
+    }
+
+    @Test
+    public void tokenizeVenmoAccount_whenCreatePaymentContextSucceeds_persitsPaymentContextIdThenStartsVenmoActivityAndSendsAnalytics() throws JSONException {
```

Comment:
Is this still right? I thought we were no longer persisting the `paymentContextId`, since we're checking for it in the return URL instead.

Reply:
Right yeah good call. We updated the code and not the test name. We can take that out.